### PR TITLE
Feature/simplify permissions

### LINKED ISSUE #2197: Simplify permissions box on add contributor
Not really a bug. Low priority.

## Proposal
Change the add contributors permissions select box to mirror the contributors' list select boxes. This is from work on #390.

## Ramifications
One less click. Unifying similar input layouts.

### LINKED ISSUE #2209: New permission default for adding new contributors.
Steps
-------
1. Go to "sharing" tab on project, click add button
2. find and add contributors

Expected
------------
Default permission is not admin.

Actual
--------
Default is admin.

Suggestion
--------
Set to read, read+write, or blank making the admin adding contributors pick one of the permissions options themselves.

I would like to discuss this.

### PR DESCRIPTION
**Pull request by Ghalko**
## Proposed

Permissions now look like PR #2196. Close #2197. Default set to "Read", close #2209.

## Repercussions
Many fewer accidental admins. Two issues at once, but the 'default permission' is a one line change.

### CODE REVIEW

File: website/static/js/contribAdder.js [hunk 1]
```diff
@@ -337,7 +320,10 @@ var AddContributorViewModel = oop.extend(Paginator, {
         return $osf.postJSON(
             nodeApiUrl + 'contributors/', {
                 users: self.selection().map(function(user) {
-                    return ko.toJS(user);
+                    var permission = user.permission().value; //removes the value from the object
+                    var tUser = JSON.parse(ko.toJSON(user)); //The serialized user minus functions
```
Comment (sloria) [inline website/static/js/contribAdder.js]: This is rendering user to JSON then parsing JSON to JS, which seems redundant.

Comment (sloria) [inline website/static/js/contribAdder.js]: Oh, I should have read the comment. You needed to remove the methods. Disregard previous comment.

File: website/static/js/contribAdder.js [hunk 2]
```diff
@@ -337,7 +320,10 @@ var AddContributorViewModel = oop.extend(Paginator, {
         return $osf.postJSON(
             nodeApiUrl + 'contributors/', {
                 users: self.selection().map(function(user) {
```
Comment (sloria) [inline website/static/js/contribAdder.js]: For browser compatibility, can you change this to use `ko.utils.arrayMap` or `$.map`?

File: website/static/js/contribAdder.js [hunk 3]
```diff
@@ -336,8 +319,11 @@ var AddContributorViewModel = oop.extend(Paginator, {
         $osf.block();
         return $osf.postJSON(
             nodeApiUrl + 'contributors/', {
-                users: self.selection().map(function(user) {
-                    return ko.toJS(user);
+                users: ko.utils.arrayMap(self.selection(), function(user) {
+                    var permission = user.permission().value; //removes the value from the object
+                    var tUser = JSON.parse(ko.toJSON(user)); //The serialized user minus functions
```
Comment (sloria) [inline website/static/js/contribAdder.js]: Hm, could you do `ko.toJS(user)`.

**Comment by sloria** [comment]
The order of permissions in the select input for the Add contributor modal is the reverse of the order in the Contributors table. Can you make this consistent?

**Comment by sloria** [comment]
Would it be much more work to also make the inputs in the Contributors table plain `select` inputs, like you did for the modal?

**Comment by sloria** [comment]
Looks good. Minor changes and this is good to merge. :rugby_football:

File: website/static/js/contribAdder.js (hunk 3 above)
Comment (Ghalko) [inline website/static/js/contribAdder.js]: @sloria 

https://github.com/knockout/knockout/issues/251

For TL;DR: Basically toJS keeps the functions, toJSON removes ko functions and an error associated with them. They talked about removing functions with toJS, but decided against it as there are people that rely on the return of functions.

**Comment by Ghalko** [comment]
The contributors table is fixed in #2571. This should have the same order, ```[read, write, admin]``` as the table fix. Less clicks for happy users!

**Comment by Ghalko** [comment]
Might fix #2459.

File: website/static/js/contribAdder.js (hunk 3 above)
Comment (sloria) [inline website/static/js/contribAdder.js]: Ah, I see. Thanks for the clarifications.

**Comment by sloria** [comment]
@Ghalko The default permissions don't seem to have been changed in the UI; still defaults to Read.

File: website/static/js/contribAdder.js [hunk 4]
```diff
@@ -222,31 +227,9 @@ var AddContributorViewModel = oop.extend(Paginator, {
             $(element).find('.contrib-button').tooltip();
         });
     },
-    setupEditable: function(elm, data) {
```
Comment (sloria) [inline website/static/js/contribAdder.js]: I believe the `bootstrap-editable` import can be removed as well.

**Comment by sloria** [comment]
Pass finished. :v:

**Comment by sloria** [comment]
Pass finished. :vibration_mode:

**Comment by GaryKriebel** [comment]
@sloria .... this was marked for 0.36 release ... did it not go it with it?

**Comment by sloria** [comment]
This was not released with 0.36. It will go with next release.

**Pull request merged**